lisa/tools/kdump.py

```
@@ -0,0 +1,155 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from lisa.tools.service import Systemctl
+from lisa.tools.echo import Echo
+
+from pathlib import PurePosixPath
+from lisa.executable import Tool
+from lisa.tools import Sed, Cat
+from lisa.operating_system import Posix, Redhat, Suse, Debian
+from typing import cast
+from assertpy import assert_that
+
+
+class Kdump(Tool):
+    # Redhat: https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/kernel_administration_guide/kernel_crash_dump_guide
+    # Ubuntu: https://ubuntu.com/server/docs/kernel-crash-dump
+    # Suse: https://www.suse.com/zh-cn/support/kb/doc/?id=000016171
+    kexec_crash = "/sys/kernel/kexec_crash_loaded"
+
+    @property
+    def command(self) -> str:
+        return "makedumpfile"
+
+    @property
+    def can_install(self) -> bool:
+        return True
+
+    def _check_exists(self) -> bool:
+        if isinstance(self.node.os, Redhat):
+            if self.node.os.package_exists("kexec-tools"):
+                return True
+        elif isinstance(self.node.os, Debian):
+            if self.node.os.package_exists("linux-crashdump"):
+                return True
+        elif isinstance(self.node.os, Suse):
+            if (
+                self.node.os.package_exists("kexec-tools")
+                and self.node.os.package_exists("kdump")
+                and self.node.os.package_exists("makedumpfile")
+            ):
+                return True
+        return False
+
+    def _install(self) -> bool:
+        if isinstance(self.node.os, Redhat):
+            self.node.os.install_packages("kexec-tools")
+        elif isinstance(self.node.os, Debian):
+            self.node.os.install_packages("linux-crashdump")
+        elif isinstance(self.node.os, Suse):
+            package_names = ["kexec-tools", "kdump", "makedumpfile"]
+            self.node.os.install_packages(package_names)
```

How about else? If it doesn't know, it should raise an exception, instead of do nothing silently.

---

Add "raise an exception" in `command `function. In the install function, we don't need to add `else`

---

It doesn't matter how other methods handle it. The logic among methods may be changed. The `else` is the guard sentence, the logic in each method should be completed.

---

Done